app/controllers/users_controller.rb

```diff
@@ -4,25 +4,29 @@ class UsersController < ApplicationController
   before_action :get_user_by_id, only: [:show, :edit, :update]
 
   def index
-    @pagy, @users = pagy(User.all)
+    @pagy, @users = pagy(User.where(activated: true))
```

Comment:
- Đưa `activated: true` vào scope em nhé
- So sánh scope vs class method

Reply:
`Scope` là 1 built-in method của module `ActiveRecord` mà khi được định nghĩa nó sẽ tạo 1 class method bằng metaprogramming để query objects. Nên scope cũng đc sử dụng giống như 1 class method khi ta cần gọi đến nó.

`Scope` có một số điểm khác biệt so với class method thông thường:
- `Scope` luôn trả về duy nhất object `ActiveRecord::Relation` còn class method thường sẽ trả về gì cũng được. Điều này có nghĩa là class method có thể sẽ trả về `nil`, gây nên nguy cơ raise exception cho query -> break program
- `Scope` có thể mở rộng được module dễ dàng bằng extensions, trong khi đó class method sẽ phải định nghĩa module riêng để extend